Cl 2856/custom pages

IvaKop opened the pull request:
# Changelog
## Technical
- Convert custom pages to React Query

IvaKop committed:
Replace get custom page

IvaKop committed:
Add delete to custom page

IvaKop committed:
Convert update

IvaKop committed:
Move util function

IvaKop committed:
Add get list hook

IvaKop committed:
Delete service

IvaKop committed:
Delete dead code

IvaKop committed:
Fix tests

IvaKop committed:
Fix mocks

IvaKop commented:
@brentguf A quick functional review on this one would be nice, as well.

alexander-cit commented:
(regression)

Not existing page is just blank http://localhost:3000/en/pages/not-existing-page
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/7171e68b-3e26-4629-b031-a40bf8c51f7c)

On master, it shows "Page not found" https://juliennesplayground.citizenlab.co/en/pages/not-existing-page
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/1ab5233d-fa99-4fc9-9f9f-c396d3b3981c)

alexander-cit commented:
(reproduced in master too)

1. Reorder some pages
2. Add more pages to the navbar

Result: the new pages are shown in the navbar, but not shown in the list.
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/5b9b83ea-1261-4b5b-9aa2-2028dfbc5164)

Expected result: the new pages are shown both in the navbar and in the list.

alexander-cit commented:
(regression)

1. Remove the image in the hero banner section
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/68e42c64-6e37-4c6c-9001-cc998c14faec)
2. Save the form

Master: the image is restored
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/42c462c5-00fc-4e97-a98e-7e3bd5d70d45)

This branch: the image is not restored. Though it is restored after reloading the page.
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/6219fdf9-128a-4566-bda5-fcd12b12831d)

alexander-cit reviewed (approved):
The code looks good! The custom page e2e tests succeed `make e2e-run-test spec=cypress/e2e/admin/pages_and_menu/`

But probably, regressions should be fixed.

alexander-cit commented:
If you'll want to test it manually again after fixes, you can find the steps here https://www.notion.so/citizenlab/Manual-testing-d50d02b57cf742a2815d00033928c622 (see "Custom pages and menu").

IvaKop committed:
Merge branch 'master' into CL-2856/custom-pages

IvaKop committed:
Fix navbar caching

IvaKop committed:
Fix page not found bug

IvaKop committed:
Invalidate navbar items on page add

IvaKop committed:
Fix banner image logic

IvaKop commented:
@alexander-cit Regarding the image banner, looking at the code, I don't think what's on master was the intended behaviour. I think you are not supposed to be able to save the form without a banner which is how I changed it to work now. 

The rest of the issues are addressed and fixed, thank you so much for the review 👍
